plum/resolvable.py

```diff
@@ -41,6 +41,18 @@ def deliver(self, obj):
 
     def resolve(self):
         if self._obj is None:
-            raise ResolutionError("Promise was not kept.")
+            raise ResolutionError(f"Promise `{self!r}` was not kept.")
         else:
             return self._obj
+
+    @property
+    def resolved(self):
+        """bool: True if the Promise has been kept."""
+        return self._obj is not None
+
+    def __repr__(self):
+        # fallback in case superclasses don't overload repr
```

Comment:
```suggestion
        # Fallback in case subclasses don't overload `__repr__`.
```

(I'm assuming you mean classes that inherit `Promise`.)